src/parser/text_lexer.ml

```diff
@@ -432,12 +435,31 @@ let rec token buf =
     let operator = Utf8.lexeme buf in
     try Hashtbl.find keywords operator with Not_found -> unknown_operator buf
   end
+  (* comment *)
   | ";;" ->
     single_comment buf;
     token buf
   | "(;" ->
     comment buf;
     token buf
+  (* custom annotation *)
+  | "(@", name ->
+    let annotid = Utf8.lexeme buf in
+    let annotid = String.sub annotid 3 (String.length annotid - 4) in
```

Comment:
Maybe check that the string has the minimum required length before using `String.sub` and raise an error if not ?